tests/iscsi/iscsi_server.pm

```diff
@@ -16,91 +16,249 @@ use strict;
+    target_service_tab;
+    target_backstore_tab;
+    wait_serial("$module_name-0", 180) || die "'yast2 iscsi-lio-server' didn't finish or exited with non-zero code";
+    # verify systemd services after configuration
+    record_info 'Systemd - after', 'Verify status of iscsi services and sockets';
+    my $service = (is_sle('>=15') ? 'targetcli.service' : 'target.service');
+    systemctl("is-active $service", expect_false => 0);
+    # create mutex for child job -> triggers start of initiator configuration
+    # setup is done client can connect
+    mutex_create('iscsi_target_ready');
+    my $children = get_children();
+    my $child_id = (keys %$children)[0];
+    # wait for child mutex, initiator is being configured
+    mutex_wait("iscsi_initiator_ready", $child_id, 'Initiator configuration in progress!');
+    # verify iscsi connections, ACL must be set!
+    record_info 'iSCSI Sessions', 'Display target sessions & settings after setup';
+    display_targets(expected => 'LOGGED_IN');
+    # initiator can continue to test iscsi drive
+    mutex_create('iscsi_display_sessions');
+    # wait idle while initiator finishes its execution
+    wait_for_children;
     type_string "killall xterm\n";
-    wait_for_children;                                                                  # run till client is done
+    wait_for_children;    # run till client is done
     $self->result('ok');
 }
 
 sub test_flags {
     return {fatal => 1};
 }
 
+sub post_fail_hook {
+    my $target_label    = $test_data->{target_conf}->{name} . '\\:' . $test_data->{target_conf}->{id};
+    my $initiator_label = $test_data->{initiator_conf}->{name} . '\\:' . $test_data->{initiator_conf}->{id};
+    select_console 'root-console';
+    display_targets;
+    unless (script_run('ls -la /sys/kernel/config/iscsi')) {
+        # show amount of normal logouts
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/fabric_statistics/iscsi_logout_stats/normal_logouts';
+        # show amount of abnormal logouts
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/fabric_statistics/iscsi_logout_stats/abnormal_logouts';
+        # show amount of active sessions
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/fabric_statistics/iscsi_instance/sessions';
+        # show ACL information
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/tpgt_1/acls/' . $initiator_label . '/info';
+        # show auth credentials and configuration
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/tpgt_1/acls/' . $initiator_label . '/tpgt_1/auth/*';
+        # show auth credentials and acls configuration
+        script_run 'cat /sys/kernel/config/target/iscsi/' . $target_label . '/tpgt_1/acls/' . $initiator_label . '/tpgt_1/acls/auth/*';
+    }
+    script_run "save_y2logs /tmp/y2logs_target.tar.bz2";
```

Comment:
Hmm, I wonder if we should execute our default steps for the yast tests? See also my comment above.